src/content/docs/en/guides/server-side-rendering.mdx

```diff
@@ -213,6 +213,24 @@ if (!isLoggedIn(cookie)) {
 </html>
 ```
 
+### `Astro.locals`
+
+The `Astro.locals` object is a user-defined object that can be used to access data from your Astro middleware or your server inside your page. This can be useful for passing per-request data like user sessions.
+
+```astro title="src/pages/account.astro" {8}
+---
+const { user } = Astro.locals;
+
+// If the user is not logged in, redirect them to the login page
+if (user) {
+  return Astro.redirect('/login');
```

Comment:
```suggestion
  return Astro.redirect('/login/');
```
It might not make a difference, but we do typically show our routes with trailing slashes because in some projects (like docs) we do need to write them. If there's no good reason *not* to, then I'd suggest this.

If there *is* a good reason not to, carry on!